pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -138,6 +138,7 @@ public class MutableSegmentImpl implements MutableSegment {
   private final RealtimeSegmentStatsHistory _statsHistory;
   private final String _partitionColumn;
   private final PartitionFunction _partitionFunction;
+  private final int _mainPartitionId;
```

Comment:
```suggestion
  private final int _consumingPartitionId;
```
Just a suggestion. Or else, add a comment there as to what you mean by "main"